internal/functions/efgs/download-batch.go

```diff
@@ -5,49 +5,115 @@ import (
 	"encoding/json"
 	"fmt"
 	efgsapi "github.com/covid19cz/erouska-backend/internal/functions/efgs/api"
+	efgsconstants "github.com/covid19cz/erouska-backend/internal/functions/efgs/constants"
 	efgsutils "github.com/covid19cz/erouska-backend/internal/functions/efgs/utils"
 	"github.com/covid19cz/erouska-backend/internal/logging"
-	"github.com/covid19cz/erouska-backend/internal/pubsub"
+	redisclient "github.com/go-redis/redis/v8"
 	keyserverapi "github.com/google/exposure-notifications-server/pkg/api/v1"
 	"io/ioutil"
 	"net/http"
+	"strconv"
 	"strings"
 	"time"
 )
 
 //DownloadAndSaveKeys Downloads batch from EFGS.
-func DownloadAndSaveKeys(ctx context.Context, m pubsub.Message) error {
-	var payload efgsapi.BatchDownloadParams
-
-	if decodeErr := pubsub.DecodeJSONEvent(m, &payload); decodeErr != nil {
-		return fmt.Errorf("Error while parsing event payload: %v", decodeErr)
-	}
+func DownloadAndSaveKeys(ctx context.Context) error {
```

Comment:
For better consistency it will be better return nothing. Error handling can be done inside of the function (e.g. like `DownloadAndSaveYesterdaysKeys`).